### PR TITLE
Clean up auto_flush; add flush_mapped to use listeners

### PR DESCRIPTION
**Pull request by guyboertje**

### CODE REVIEW

File: lib/logstash/codecs/identity_map_codec.rb [hunk 1]
```diff
@@ -164,6 +164,16 @@ def encode(event, identity = nil)
     stream_codec(identity).encode(event)
   end
 
+  def flush_mapped(listener)
```
Comment (colinsurprenant) [inline lib/logstash/codecs/identity_map_codec.rb:167]: I don't see where this `flush_mapped` is being used?!

File: lib/logstash/codecs/multiline.rb [hunk 2]
```diff
@@ -216,6 +216,8 @@ def flush(&block)
   end
 
   def auto_flush
+    return if @last_seen_listener.nil?
```
Comment (colinsurprenant) [inline lib/logstash/codecs/multiline.rb:219]: would it be worth introducing a `using_listeners?` method or similar?

```
def using_listeners?
  !@last_seen_listener.nil?
end
```

that would have the benefit of making it a little more self-documenting?

Comment (guyboertje) [inline lib/logstash/codecs/multiline.rb:219]: OK I buy that.

File: lib/logstash/codecs/identity_map_codec.rb (hunk 1 above)
Comment (guyboertje) [inline lib/logstash/codecs/identity_map_codec.rb:167]: Its used in the file input (or any input that wraps the given codec in an IMC)
Needs a different PR for file input.

File: lib/logstash/codecs/multiline.rb (hunk 2 above)
Comment (colinsurprenant) [inline lib/logstash/codecs/multiline.rb:219]: SOLD! ke-tching!

File: lib/logstash/codecs/identity_map_codec.rb (hunk 1 above)
Comment (colinsurprenant) [inline lib/logstash/codecs/identity_map_codec.rb:167]: ok!

**Comment by guyboertje** [comment]
This PR is redundant - I will include these changes into a new PR that fixes #28

**Pull request closed without merging**